massifs/massifcontextverified.go

```diff
@@ -213,24 +229,26 @@ func (mc *MassifContext) verifyContext(
 	// check the remote log is consistent with the log portion they have locally
 	// before replicating the new data.
 	if options.trustedBaseState != nil {
-		rootB2, err := mc.CheckConsistency(*options.trustedBaseState)
+
+		ok, _, err = mmr.CheckConsistency(
+			mc, sha256.New(),
+			options.trustedBaseState.MMRSize,
+			mc.RangeCount(),
+			options.trustedBaseState.Peaks)
 		if err != nil {
 			return nil, err
 		}
-		// rootB above will be nil if the new state is the same as the trusted
-		// state, in which case there is no value in getting the root in order
-		// to do the compare.
-		if rootB != nil && !bytes.Equal(rootB, rootB2) {
+		if !ok {
 			return nil, fmt.Errorf(
-				"%w: the root produced for the trusted base state doesn't match the root produced for the seal state fetched from the log",
-				ErrInconsistentState)
+				"%w: the accumulator produced for the trusted base state doesn't match the root produced for the seal state fetched from the log",
+				mmr.ErrConsistencyCheck)
 		}
 	}
 
 	return &VerifiedContext{
 		MassifContext:   *mc,
 		Sign1Message:    *msg,
 		MMRState:        state,
-		ConsistentRoots: rootB,
+		ConsistentRoots: peaksB,
```

Comment:
I think there are alot of terms floating around for what is basically the accumulator? Would it be worth adding an entry to the cheatsheet and in a separate PR making the names consistent.

Things like peakBagRHS, ConsistentRoots, peaksA, peaksB...etc (i know consistent roots are not always accumulators by their definition)

Reply:
Yeah, I think we do need to do that pass. I think a mmriver redux pr is in order to follow up.